File: src/main/java/br/com/useinet/finance/model/Conta.java
package br.com.useinet.finance.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;

@Entity
@Table(name = "contas")
public class Conta {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nome;
    private Double saldo;

    @ManyToOne
    private Usuario usuario;

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public String getNome() { return nome; }
    public void setNome(String nome) { this.nome = nome; }
    public Double getSaldo() { return saldo; }
    public void setSaldo(Double saldo) { this.saldo = saldo; }
    public Usuario getUsuario() { return usuario; }
    public void setUsuario(Usuario usuario) { this.usuario = usuario; }
}

File: src/main/kotlin/br/com/useinet/finance/model/Conta.kt
package br.com.useinet.finance.model

import br.com.useinet.finance.config.AesAttributeConverter
import jakarta.persistence.*

@Entity
@Table(name = "contas")
open class Conta {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

    var nome: String? = null

    var saldo: Double? = null

    @Convert(converter = AesAttributeConverter::class)
    @Column(name = "numero_conta", length = 512)
    var numeroConta: String? = null

    @Convert(converter = AesAttributeConverter::class)
    @Column(length = 512)
    var agencia: String? = null

    @ManyToOne
    var usuario: Usuario? = null
}
